library/ssl_tls.c

```diff
@@ -3640,9 +3589,74 @@ static int ssl_parse_record_header( mbedtls_ssl_context *ssl )
             return( MBEDTLS_ERR_SSL_UNEXPECTED_RECORD );
         }
```

Comment:
Not necessarily related to this patch, but: how do we know that there are enough bytes in the input buffer for `mbedtls_ssl_dtls_replay_check`? It reads 8 bytes from `ssl->in_ctr`.

Reply:
@gilles-peskine-arm `ssl_parse_record_header` is called [from here](https://github.com/ARMmbed/mbedtls/blob/development/library/ssl_tls.c#L3873), so at least `13` record header bytes are available in case of DTLS, and the 8 bytes starting at `ssl->in_ctr` [belong to that](https://github.com/ARMmbed/mbedtls/blob/development/library/ssl_tls.c#L5635).

Reply:
Good, thanks for checking!